### ISSUE TITLE
[Bugs] Some bugs/issues

### ISSUE THREAD

**Issue by mantazzo** [masterplane8/F1-League-Management#4]
Hello,
First of all, I want to thank you for the effort to make this table - it's quite powerful! However, on my first proper use I discovered that there are a few small bugs/issues in the table:

- Decimal point. Since in my country a comma (,) is used as a decimal symbol, I had issues when trying to fill in the table (Results Tables) with the dots (.) - I initially fixed this with using commas, however I changed the decimal numerator in the settings to not use system settings. So yeah, small bug (may be worth mentioning to use a default decimal symbol instead of a usual dot), but nothing serious.
- In Results Tables, Time from leader doesn't support gap over a minute big - may be worth mentioning "if the gap is over a minute, use seconds and add 60 to that number".
- In Race Reports, the interval is not correctly updated for more than 1 lap.
Potential solution that worked (example for 1 lap): `IF(AND(RaceTimesRaw!I15-MIN(RaceTimesRaw!I$4:I$43)>3500;RaceTimesRaw!I15-MIN(IF(RaceTimesRaw!I$4:I$43>4500;RaceTimesRaw!I$4:I$43)));"1 Lap";`
Basically, I edited the final MIN section because for some reason it was always NOT 0 (for some reason the original one wouldn't detect value 5000), but with this additional IF, now it works for now.
- Driver Statistics seem to be mostly broken, but then again, I have done only one race so far to write into the table. I'll have a more thorough look to see what may be wrong.

**Comment by mantazzo** [comment]
Following up with some potential improvements/fixes to consider:

- ResultsRaw (hidden). I made a few improvements which seem to work and this may look actually better. In Driver Race Finishing Positions section, I changed the finishing position of DNF/DSQ to 21 because Position 20 CAN be taken by someone if all 20 drivers start and finish; I also improved Average Race Finish Position formula to calculate only races which were started and finished (helped by finishing number value change to 21):

<details>
<summary>Details</summary>

Driver Race Finishing Positions formula:
=IF('Results Tables'!$G$5="Activated";VALUE(IF('Results Tables'!$H8="DNF";21;IF('Results Tables'!$H8="DSQ";21;IF('Results Tables'!$H8="DNP";0;IF('Results Tables'!$H8="BAN";0;IF('Results Tables'!$H8="-";0;'Results Tables'!$H8))))));0)

Average Race Finish Position formula:
=IFERROR(ROUND(SUMIF(AC4:AZ4;"<>21")/(BA4-COUNTIF(AC4:AZ4;21));2);"-")

Quick explanation - SUMIF sums all values of certain range which fit the criteria (sums all fields which are NOT 21 in this case), COUNTIF calculates the number of fields in the range which fit the criteria (counts all fields which ARE 21 in this case). In here, this means that the fields are added and divided by the races scored (Races Participated In, minus Races With DNF/DSQ).

</details>

More potential improvements may be possible after looking at Driver Qualy Results section, but for now I'll stop here.

- Driver Statistics. Table seemed to be mostly broken (at least it seemed so) as it was using mostly incorrect values for columns in the INDEX functions.
-- Fixes here seem to be as follows:
-- Races Started: Column 53 (set as 49 by default)
-- Avg. Race Result: Column 54 (from 50)
-- Race Wins: Column 86 (from 80)
-- Fastest Laps: Column 85 (from 79)
-- Avg. Qualy Result: Column 27 (from 25)
-- Poles: Column 28 (from 26)
-- Penalty Points: Column 84 (from 78)

**Comment by masterplane8** [comment]
I know see what your saying about setting to 21. Adding on to your point, it means that the statistic that shows average finishing position wont include DNF/DSQ, which would solve the problem I stated earlier. I'll change that up for the next update.

**Comment by mantazzo** [comment]
Yeah I thought using 21 would make sense, because it's a value that will never be used in results (1-20 for finishing, 0 for DNP, so for DNF/DSQ it makes sense to use some other value to make it different), and using the SUMIF/COUNTIF makes Average calculations easier. I guess a value instead of 21 could be used if you prefer...

On another note, I see that in "League Management" section, the Table for Reserve Drivers would not update proper - I checked the formula, and it was referencing some field which seemed to be completely empty, so I adjusted it a bit:

<details>
<summary>The Change</summary>

=IF('Reserve Drivers'!C4="";"-";'Reserve Drivers'!C4)

</details>

As for ideas with Qualy table part, I was thinking maybe it would be possible to simplify the Race table section by removing the DNF/DSQ check and adding them to the table, however it would break the "races participated in" calculation, so that's out of the question currently.

Finally, I'm not sure how much is this viable, but could you maybe add "the countback" into consideration for driver standings? As an example (only one race results so far), currently I have a driver who finished P10 and got a fastest lap, and in the driver standings he is shown in P9 because he has the same amount of points as the person who finished P9... Not sure if a proper countback would be difficult to make, but just a small idea.
